protocol.md

```diff
@@ -180,7 +181,11 @@ A *virtual device* can not be revoked.
 
 The only way to add more *device*s, after the first one created during [user registration](#user-registration), is to sign a new `device_creation` block containing the new device's keys with a *user*'s existing device.
 
-For this purpose, during [user registration](#user_registration) *Tanker Core* creates a *virtual device* and pushes its `device_creation` block. This *device* is not a "physical" one, it has no [Local Encrypted Storage]. Instead, the [Device Signature Key Pair] and the [Device Encryption Key Pair] are serialized as an opaque token, we call it the [Verification Key]. The [Verification Key] is encrypted with the [User Secret] and stored on the *Tanker server*.
+For this purpose, during [user registration](#user_registration) *Tanker Core* creates a *virtual device* and pushes its `device_creation` block. This *device* is not a "physical" one, it has no [Local Encrypted Storage]. Instead, the [Device Signature Key Pair] and the [Device Encryption Key Pair] are serialized as an opaque token, we call it the [Verification Key].
+
+There are two types of verification methods: end-to-end (E2E) methods, and non-E2E methods.
+The difference is that for non-E2E methods, the [Verification Key] is encrypted with the [User Secret] and stored on the *Tanker server*, while for E2E methods it is encrypted with the [User Encryption Key Pair] and the E2E method's secret value (e.g. an E2E passphrase) before being stored on the *Tanker server*.
+The reason for this difference is that the [User Secret] can be accessed by parties others than the end-users themselves, namely the *application*, so it is not strictly end-to-end. End-to-end verification methods offer stricter guarantees and do not allow for any possibility of recovery if the verification method's secret value (e.g. passphrase) is lost.
```

Comment:
```suggestion
The reason for this difference is that the [User Secret] can be accessed by parties others than the end-users themselves, namely the *application*, so it is not strictly end-to-end. End-to-end verification methods offer stricter guarantees but do not allow any possibility of recovery if the verification method's secret value (e.g. passphrase) is lost.
```

Reply:
Added the but, but I think the for is correct..?